engine/crates/integration-tests/tests/federation/basic/enums.rs

```diff
@@ -0,0 +1,31 @@
+use engine_v2::Engine;
+use graphql_mocks::TeaShop;
+use integration_tests::{federation::EngineV2Ext, runtime};
+
+#[test]
+fn inaccessible_enum_values_are_inaccessible() {
+    let response = runtime().block_on(async move {
+        let engine = Engine::builder().with_subgraph(TeaShop::default()).build().await;
+
+        engine.post("query { recommendedTeas { id name style } }").await
+    });
+
+    insta::assert_json_snapshot!(response, @r#"
+    {
+      "data": {
+        "recommendedTeas": [
+          {
+            "id": 0,
+            "name": "Earl Grey",
+            "style": "BLACK"
+          },
+          {
+            "id": 7,
+            "name": "Tuóchá",
+            "style": null
```

Comment:
Style isn't nullable in your schema so is setting this to null correct?  Would have kind of expected it to behave like errors and bubble up to the nearest nullable or something like that.

Reply:
Good point. It won't emit an error but we should bubble up the nullability. That means I probably haven't picked the right place for that change in response processing.